### DISCUSSION TITLE
freemind: cannot run shell script in user home

### DISCUSSION THREAD

**Discussion by gitsteff** [netblue30/firejail#5260] [Q&A] [answered]
Hi.
I would like to call Freemind like that because I want to make some changes to the profile and so I copied the original profile to ~/.config/firejail/:

firejail --profile=~/.config/firejail/freemind.profile /home/xyz/Software/freemind/FreeMind/freemind.sh

Same applies when I try to start it this way.
firejail --profile=/etc/firejail/freemind.profile /home/xyz/Software/freemind/FreeMind/freemind.sh

When I start it like this:
firejail /home/xyz/Software/freemind/FreeMind/freemind.sh
the app starts and can be seen when running
firejail --list

I can confirm that /home/xyz/Software/freemind/FreeMind/freemind.sh exists and file permissions are set up.

What am I doing wrong?
Thanks.

**Comment by rusty-snake** [comment, marked as answer]
Solution: https://github.com/netblue30/firejail/discussions/5260#discussioncomment-3182665

> Because it's a shell script you can […] start it as `firejail --profile=freemind /bin/bash /home/xyz/Software/freemind/FreeMind/freemind.sh` […].

---

1. ~What changes did you done?~
2. Why do you copy the profile rather than using a .local?

---

You need to allow executing files in `$HOME` if you install freemind in it.

```
ignore apparmor
ignore noexec ${HOME}
```

**Reply by gitsteff** [reply]
Just for better understanding:
If there are conflicting options - say local has net none and .profile has net=eth0 - which one will be triggered?

**Reply by rusty-snake** [reply]
Depends on the command.

For `net`, you can be sure that `net eth0` will never be in a profile.

**Comment by gitsteff** [comment]
ignore noexec ${HOME}
did the trick but opens the door wide.

But I can not be more specific, say
`ignore noexec /home/xyz/Software/freemind/FreeMind-v1.1.0_Beta2/ `
which causes the error to be triggered again.

Do you see any solution for just allowing this wrapper shell script just at this one location?
Thanks.

**Reply by rusty-snake** [reply]
With firejail, no.

**Reply by rusty-snake** [reply]
Because it's a shell script you can also start it as `firejail --profile=freemind /bin/bash /home/xyz/Software/freemind/FreeMind/freemind.sh` and keep `noexec ${HOME}`.

**Reply by gitsteff** [reply]
Your last comment

> Because it's a shell script you can also start it as firejail --profile=freemind /bin/bash /home/xyz/Software/freemind/FreeMind/freemind.sh and keep noexec ${HOME}.
did the trick.

Well, also this is the solution to not allow execution in /home but start specific apps in /home: Writing a wrapper script will enable the behaviour I was asking for...

Great support, thanks.
